extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriColor.java

```diff
@@ -124,9 +124,9 @@ public static TradfriColor fromCie(int xyX, int xyY, int xyBrightness) {
         // green = green <= 0.0031308 ? 12.92 * green : (1.0 + 0.055) * Math.pow(green, (1.0 / 2.4)) - 0.055;
         // blue = blue <= 0.0031308 ? 12.92 * blue : (1.0 + 0.055) * Math.pow(blue, (1.0 / 2.4)) - 0.055;
 
-        int redRounded = (int) Math.round(red * 255.0);
-        int greenRounded = (int) Math.round(green * 255.0);
-        int blueRounded = (int) Math.round(blue * 255.0);
+        int redRounded = ensureRgbRange((int) Math.round(red * 255.0));
```

Comment:
I wonder if this is the right place for the fix. Shouldn't we make sure that red/green/blue is not negative and not above 1? At least the later is already checked in lines 108-120 so you would only need to check for negative numbers.

Reply:
Thanks for your comment, that would be indeed better.
I'm going to change this and rebase on master, as there are now duplicate method names in the tests I guess.

Reply:
Done.